### DISCUSSION TITLE
Can the optimizer state be saved to resume training later?

### DISCUSSION THREAD

**Discussion by JonLangdon** [xamarin/TorchSharp#561] [Q&A]
Is there a way to save the internal state of the optimizer (ie. the state_dict()) as can be done in pytorch? I have a long running training session and would like to be able to stop and restart training as needed. I am using Adam optimizer in for my model.

**Comment by NiklasGustafsson** [comment]
@JonLangdon -- not at this time. Currently, we can only save the model state. It's been on my mind, but not yet done. 

I'll open an issue on this.
